cfme/tests/intelligence/reports/test_validate_chargeback_report.py

```diff
@@ -202,24 +209,36 @@ def resource_usage(vm_ownership, db, provider, ssh_client_modscope):
             "average_disk_io": average_disk_io}
 
 
-def query_rate(db, provider, metric, description, rate_type):
+def resource_cost(db, provider, metric, usage, description, rate_type):
     # Query the DB for Chargeback rates
     tiers = db['chargeback_tiers']
     details = db['chargeback_rate_details']
     rates = db['chargeback_rates']
+    list_of_rates = []
+
+    def add_rate(tiered_rate):
+        list_of_rates.append(tiered_rate)
 
     with db.transaction:
-        providers = (
-            db.session.query(tiers.variable_rate).
+        result = (
+            db.session.query(tiers).
             join(details, tiers.chargeback_rate_detail_id == details.id).
             join(rates, details.chargeback_rate_id == rates.id).
             filter(details.metric == metric).
             filter(rates.rate_type == rate_type).
-            filter(rates.description == description)
+            filter(rates.description == description).all()
         )
-    rate = db.session.query(tiers).filter(tiers.variable_rate.in_(
-        providers.subquery())).first().variable_rate
-    return rate
+    for row in result:
+        tiered_rate = {var: getattr(row, var) for var in ['variable_rate', 'fixed_rate', 'start',
+            'finish']}
+        add_rate(tiered_rate)
+
+    # Check what tier the usage belongs to and then compute the usage cost based on Fixed and
+    # Variable Chargeback rates.
+    for d in list_of_rates:
+        if usage >= d['start'] and usage < d['finish']:
+            cost = (d['variable_rate'] * usage * 24) + (d['fixed_rate'] * 24)
```

Comment:
**Optional**
Consider putting 24 in a constant so it has some meaning (guessing its hours in a day?)